File: apps/server/app/schemas/report.py
from datetime import date
from typing import List, Optional

from pydantic import BaseModel, Field


class SessionFocusItem(BaseModel):
    date: date
    note: str


class WeeklyReportResponse(BaseModel):
    week_start: date
    week_end: date
    week_distance_m: int
    week_duration_min: int
    week_calorie_kcal: int
    calorie_caption: str = "또래 평균 체중 기준 추정치예요"
    cumulative_distance_m: int
    session_focus_list: List[SessionFocusItem] = Field(default_factory=list)
    apply_tip: Optional[str] = None
    key_points: Optional[List[str]] = None
    rating: Optional[int] = None

    class Config:
        from_attributes = True


class RatingRequest(BaseModel):
    rating: int = Field(..., ge=1, le=5)


class RatingResponse(BaseModel):
    rating: int

File: apps/mobile/src/api/report.ts
import { apiClient } from './client';

export interface SessionFocusItem {
  date: string;
  note: string;
}

export interface WeeklyReportResponse {
  week_start: string;
  week_end: string;
  week_distance_m: number;
  week_duration_min: number;
  week_calorie_kcal: number;
  calorie_caption: string;
  cumulative_distance_m: number;
  session_focus_list: SessionFocusItem[];
  apply_tip: string | null;
  key_points: string[] | null;
  rating: number | null;
}

export interface RatingResponse {
  rating: number;
}

export const reportApi = {
  getWeeklyReport: async (classId: number, studentId: number, date: string) => {
    const response = await apiClient.get<WeeklyReportResponse>(
      `/report/${classId}/${studentId}/${date}`,
    );
    console.log(response);
    return response.data;
  },

  submitRating: async (classId: number, studentId: number, date: string, rating: number) => {
    const response = await apiClient.post<RatingResponse>(
      `/report/${classId}/${studentId}/${date}/rating`,
      { rating },
    );
    console.log(response);
    return response.data;
  },
};
